zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,2368 @@
+                        meta.query_advice(byte_value, Rotation::cur()) - 247.expr(),
+                    );
+                    cb.require_zero(
+                        "length_acc == 0",
+                        meta.query_advice(length_acc, Rotation::cur()),
+                    );
+                },
+            );
+
+            // if tag_index == tag_length && tag_length == 1
+            cb.condition(
+                is_prefix(meta) * tindex_eq_tlength.clone() * tlength_eq.clone(),
+                |cb| {
+                    cb.require_equal("191 < value", value_gt_191.is_lt(meta, None), 1.expr());
+                    cb.require_equal("value < 248", value_lt_248.is_lt(meta, None), 1.expr());
+                    cb.require_equal(
+                        "length_acc == value - 0xc0 (1)",
+                        meta.query_advice(length_acc, Rotation::cur()),
+                        meta.query_advice(byte_value, Rotation::cur()) - 192.expr(),
+                    );
+                },
+            );
+
+            // if tag_index < tag_length && tag_length > 1
+            cb.condition(
+                is_prefix(meta) * tindex_lt_tlength.clone() * tlength_lt.clone(),
+                |cb| {
+                    cb.require_equal(
+                        "length_acc == (length_acc::prev * 256) + value",
+                        meta.query_advice(length_acc, Rotation::cur()),
+                        meta.query_advice(length_acc, Rotation::prev()) * 256.expr()
+                            + meta.query_advice(byte_value, Rotation::cur()),
+                    );
+                },
+            );
+
+            //////////////////////////////////////////////////////////////////////////////////////
+            ////////////////////////////////// RlpTxTag::Nonce ///////////////////////////////////
+            //////////////////////////////////////////////////////////////////////////////////////
+            cb.condition(is_nonce(meta), |cb| {
+                // tag_index < 10
+                cb.require_equal(
+                    "tag_index < 10",
+                    tag_index_lt_10.is_lt(meta, None),
+                    1.expr(),
+                );
+
+                // tag_index >= 1
+                cb.require_zero(
+                    "1 <= tag_index",
+                    not::expr(or::expr([tindex_lt.clone(), tindex_eq.clone()])),
+                );
+            });
+
+            // if tag_index == tag_length && tag_length == 1
+            cb.condition(
+                is_nonce(meta) * tindex_eq_tlength.clone() * tlength_eq.clone(),
+                |cb| {
+                    cb.require_equal("value < 129", value_lt_129.is_lt(meta, None), 1.expr());
+                    cb.require_equal(
+                        "value == value_acc",
+                        meta.query_advice(byte_value, Rotation::cur()),
+                        meta.query_advice(rlp_table.value_acc, Rotation::cur()),
+                    );
```

Comment:
Under this condition, value could also be equal to `0x80` which means that the decoded value should be 0 instead of `0x80`. Therefore, in this case `rlp_table.value_acc == 0`